experimenter/evaluate_pipeline_new.py

```diff
@@ -0,0 +1,282 @@
+    args.extend(('--input', input))
+    args.extend(('--output-run', output_run_path))
+    args.extend(('--data-random-seed', data_random_seed))
+
+    if (input_run):
+        if (not isinstance(input_run, str)):
+            raise TypeError('\'{}\' param not of type \'{}\''.format('input_run','str'))
+        if (not os.path.isfile(input_run) and input_run != '-'):
+            raise ValueError('\'{}\' param invalid: {file_path, \'-\'}'.format('input_run'))
+        # TODO: input_run validation
+        pass
+
+    if (data_params):
+        if (not isinstance(data_params, List)):
+            raise TypeError('\'{}\' param not of type \'{}\''.format('data_params','List'))
+        for data_param in data_params:
+            args.extend(('--data-param', data_param[0], data_param[1]))
+
+    if (data_pipeline):
+        if (not isinstance(data_pipeline, str)):
+            raise TypeError('\'{}\' param not of type \'{}\''.format('data_pipeline','str'))
+        if (not os.path.isfile(data_pipeline) and not is_valid_uuid(data_pipeline)):
+            raise ValueError('\'{}\' param not a file path or pipeline ID'.format('data_pipeline'))
+        args.extend(('--data-pipeline', data_pipeline))
+
+    if (scoring_pipeline):
+        if (not isinstance(scoring_pipeline, str)):
+            raise TypeError('\'{}\' param not of type \'{}\''.format('scoring_pipeline','str'))
+        if (not os.path.isfile(scoring_pipeline) and not is_valid_uuid(scoring_pipeline)):
+            raise ValueError('\'{}\' param not a file path or pipeline ID'.format('scoring_pipeline'))
+        args.extend(('--scoring-pipeline', scoring_pipeline))
+
+    if (metric):
+        if (not isinstance(metric, str)):
+            raise TypeError('\'{}\' param not of type \'{}\''.format('metric','str'))
+        # TODO: set of valid metric args?
+        args.extend(('--metric', metric))
+
+    if (scoring_params):
+        if (not isinstance(scoring_params, List)):
+            raise TypeError('\'{}\' param not of type \'{}\''.format('scoring_params','List'))
+        for scoring_param in scoring_params:
+            args.extend(('--scoring-param', scoring_param[0], scoring_param[1]))
+
+    if (scores):
+        args.extend(('--scores', scores_path))
+
+    if (scoring_random_seed):
+        if (not isinstance(scoring_random_seed, int)):
+            raise TypeError('\'{}\' param not of type \'{}\''.format('scoring_random_seed','int'))
+        args.extend(('--scoring-random-seed', scoring_random_seed))
+
+    if (data_split_file):
+        if (not isinstance(data_split_file, str)):
+            raise TypeError('\'{}\' param not of type \'{}\''.format('data_split_file','str'))
+        if (data_split_file != '-' and not os.path.isfile(data_split_file)):
+            raise ValueError('\'{}\' param invalid value: {file_path, \'-\'}'.format('data_split_file'))
+        args.extend(('--data-split-file', data_split_file))
+
+    cli.main(args)
```

Comment:
maybe let's just call the upload function at the end of this function. eventually the d3m cli will have this functionality, so we won't have to refactor as much when that functionality is available.

Reply:
okay. if you know, where is the `upload` function in d3m?

Reply:
That's part of what I am trying to say. D3M does not yet have an upload function. It will eventually be CLI arg in the runtime. Once it exists, we can add it to the `args` variable you have defined here. Until then, we should just call our functionality right after `cli.main(args)`.